src/adjoint/adjointUtils.F90

```diff
@@ -179,6 +179,9 @@ subroutine setupStateResidualMatrix(matrix, useAD, usePC, useTranspose, &
 
             ! Very important to use only Second-Order dissipation for PC
             lumpedDiss = .True.
+            ! We also do not apply acoustic scaling because the dissipation stabilizes ILU
```

Comment:
are you saving the `acousticScaleFactor` here and avoiding any scaling here? if the `acousticScaleFactor` is set to 1 here without saving, will it propagate to the other functions where it is used?

Reply:
`acousticScaleFactor` is saved and then restored to the user value before exiting this routine. This is handled the same way as the lumped dissipation.

Reply:
I added some more comments as we discussed @anilyil